astropy/utils/data.py

```diff
@@ -1019,90 +1019,103 @@ def download_file(remote_url, cache=False, show_progress=True, timeout=None):
     else:
         url_key = remote_url
 
-    try:
-        if cache:
-            # We don't need to acquire the lock here, since we are only reading
-            with _open_shelve(urlmapfn, True) as url2hash:
-                if url_key in url2hash:
-                    return url2hash[url_key]
+    url_key_list = url_key.split(',')
```

Comment:
On the other hand, having it as a list makes things much much more complicated in upstream functions, as they cannot simply test for it being a url like something, but rather have to deal with the loop.